engine/src/main/java/org/pentaho/di/www/SlaveServerConfig.java

```diff
@@ -107,27 +107,15 @@ public SlaveServerConfig() {
     databases = new ArrayList<DatabaseMeta>();
     slaveSequences = new ArrayList<SlaveSequence>();
     automaticCreationAllowed = false;
-    metaStore = new DelegatingMetaStore();
-    // Add the local Hitachi Vantara MetaStore to the delegation.
-    // This sets it as the active one.
-    //
-    try {
-      XmlMetaStore localStore = new XmlMetaStore( MetaStoreConst.getDefaultPentahoMetaStoreLocation() );
-      metaStore.addMetaStore( localStore );
-      metaStore.setActiveMetaStoreName( localStore.getName() );
-    } catch ( MetaStoreException e ) {
-      LogChannel.GENERAL.logError( "Unable to open local Pentaho meta store from [" + MetaStoreConst.getDefaultPentahoMetaStoreLocation() + "]", e );
-      // now replace this with an in memory metastore.
-      //
-      try {
-        MemoryMetaStore memoryStore = new MemoryMetaStore();
-        memoryStore.setName( "Memory metastore" );
-        metaStore.addMetaStore( memoryStore );
-        metaStore.setActiveMetaStoreName( memoryStore.getName() );
-      } catch ( MetaStoreException e2 ) {
-        throw new RuntimeException( "Unable to add a default memory metastore to the delegating store", e );
+    metaStoreSupplier = () -> {
+      IMetaStore metastore = MetaStoreConst.getDefaultMetastoreSupplier().get();

```

Comment:
This removes two exception handling block and error message from log. In the new code, how is the error handled? Did we test this scenario by comparing the behavior in existing code and in the updated code to make sure we are consistent

Reply:
Most of the logging will have to be in MetastoreLocatorImpl. 
We could put in a log message that we're using the memory metastore here. 

There are no more exceptions being thrown, though. Those were only coming from DelegatingMetastore, which isn't needed anymore.

Reply:
OK, so looking more closely, neither of the failure cases here that used to log can still happen, or at least, can't be detected here. 

If the local metastore is attempted and fails in the MetastoreLocator, this code can't tell. 

The second log message was only based on exceptions thrown by DelegatingMetaStore, and the methods never actually threw anything, even though they declared that they could, so it could never have failed before.

Reply:
Ah, we can still log when we use the memory metastore, it just won't have an exception. I'll add that.